src/highdicom/io.py

```diff
@@ -353,7 +360,12 @@ def _check_file_format(self, fp) -> Tuple[bool, bool]:
             If the file object does not represent a DICOM Part 10 file
 
         """
-        file_meta = read_file_meta_info(str(self._filename))
+        if self._filename is None:
+            file_meta = dcmread(fp).file_meta
+            fp.seek(0)
+        else:
+            file_meta = read_file_meta_info(str(self._filename))
```

Comment:
Let's only read the File Meta Information and avoid reading the entire file object (untested):

```python
        if self._filename is None:
            def stop_when(tag: pydicom.tag.BaseTag, vr: str, length: int) -> bool:
                if tag.group > 2:
                    return True
                return False
            file_meta = dcmread(fp, stop_when=stop_when).file_meta
            fp.seek(0)
        else:
            file_meta = read_file_meta_info(str(fp.name))
```

Reply:
Implemented in https://github.com/ImagingDataCommons/highdicom/pull/223/commits/d2eb66126facab94215377378783113506d770d5